PodcastGenerator/core/episodes.php

```diff
@@ -57,34 +95,44 @@ function getEpisodes($category = null, $_config)
-                        'category3PG' => $item->categoriesPG->category3PG
-                    ],
-                    'keywordsPG' => $item->keywordsPG,
-                    'explicitPG' => $item->explicitPG,
-                    'authorPG' => [
-                        'namePG' => $item->authorPG->namePG,
-                        'emailPG' => $item->authorPG->emailPG
-                    ],
-                    'fileInfoPG' => [
-                        'size' => $item->fileInfoPG->size,
-                        'duration' => $item->fileInfoPG->duration,
-                        'bitrate' => $item->fileInfoPG->bitrate,
-                        'frequency' => $item->fileInfoPG->frequency
-                    ],
-                    'filename' => $episode,
-                    'fileid' => pathinfo($_config['absoluteurl'] . $_config['upload_dir'] . $episode, PATHINFO_FILENAME),
-                    'moddate' => date('Y-m-d', filemtime($_config['absoluteurl'] . $_config['upload_dir'] . $episode))
-                ]
-            ];
+            $append_array = arrayEpisode($item, $episode, $_config);
+            array_push($episodes_data, $append_array);
+        }
+    }
+    unset($_config);
+    return $episodes_data;
+}
+
+function searchEpisodes($name = "", $_config)
+{
+    $name = strtolower($name);
+    $episodes_mtimes = setupEpisodes($_config);
+    // Check if name is a category and replace
+    $cats_xml = simplexml_load_file('categories.xml');
+    foreach ($cats_xml as $item) {
+        if ($name === strtolower($item->description[0])) {
+            $name = strval($item->id[0]);
+        }
+    }
+    // Get XML data for the episodes of interest.
+    $episodes_data = array();
+    for ($i = 0; $i < sizeof($episodes_mtimes); $i++) {
+        $episode = $episodes_mtimes[$i][0];
+        // We need to get the CDATA in plaintext.
+        $xml_file_name = pathinfo($_config['absoluteurl'] . $_config['upload_dir'] . $episode, PATHINFO_FILENAME) . '.xml';
+        $xml = simplexml_load_file($_config['absoluteurl'] . $_config['upload_dir'] . $xml_file_name, null, LIBXML_NOCDATA);
+        foreach ($xml as $item) {
+            if (strpos(strtolower($item->titlePG), $name) === false
+                && strpos(strtolower($item->shortdescPG[0]), $name) === false
+                && strpos(strtolower($item->longdescPG[0]), $name) === false
+                && strpos(strtolower($item->categoriesPG->category1PG[0]), $name) === false
+                && strpos(strtolower($item->categoriesPG->category2PG[0]), $name) === false
+                && strpos(strtolower($item->categoriesPG->category3PG[0]), $name) === false
+                && strpos(strtolower($item->keywordsPG[0]), $name) === false
+                && strpos(strtolower($item->authorPG->namePG[0]), $name) === false) {
+                continue;
+            }
+            $append_array = arrayEpisode($item, $episode, $_config);
```

Comment:
Remove that line and use that:
```php
array_push($episodes_data, arrayEpisode($item, $episode, $_config));
```

Reply:
Fixed, in both places this method is used.